### ISSUE TITLE
Create extension to run pyinstrument

### ISSUE THREAD

**Issue by jkimbo** [strawberry-graphql/strawberry#1308]
We should create an extension that allows you to profile your request using https://github.com/joerick/pyinstrument

Should probably an argument that lets you specify the path for the profile report.

Example API:

```python
schema = strawberry.Schema(
	Query,
	extensions=[
		PyInstrument(report_path="./my-report.html")
	]
```

**Comment by jkimbo** [comment]
@chinmaym07 that would be great!

So you can use one of the other extensions as a base to start with: https://github.com/strawberry-graphql/strawberry/blob/main/strawberry/extensions/add_validation_rules.py and make sure you take a look at the docs to see what hooks you can use: https://strawberry.rocks/docs/guides/custom-extensions#request (the request hooks are probably the ones to use here).

Also we have a template for documenting the extension: https://github.com/strawberry-graphql/strawberry/blob/main/docs/extensions/_template.md

Let me know if you encounter any issues.

**Comment by Helithumper** [comment]
I'm looking into this

**Comment by jkimbo** [comment]
Fixed in #2727 (thanks @Helithumper !)
